core/command.go

```diff
@@ -1,13 +1,57 @@
 package core
 
-import "errors"
+import (
+	"errors"
+	"sync"
+)
 
 // State represents system configuration after processing user input
 type State struct {
 	ancestors ancestors
 	Folder    *File
 	Selected  int
 	history   map[*File]int // history of all selected postions
+	*FileMarker
```

Comment:
The state should be immutable and we'll always generate new one by defensively copying the old state. Thanks to that we don't have to worry about concurrency, because the only thing that is generating the new state is `core.StartProcessing` which transforms command into a new state.